File: examples/indexer/go/SearchTransactionsPaging.go
package main

import (
	"context"
	"encoding/json"
	"fmt"

	"github.com/algorand/go-algorand-sdk/client/v2/indexer"
)

// indexer host
const indexerAddress = "http://localhost:8980"
const indexerToken = ""

// query parameters
var nextToken = ""
var numTx = 1
var numPages = 1
var minAmount uint64 = 100000000000000
var limit uint64 = 2

func main() {

	// Create an indexer client
	indexerClient, err := indexer.MakeClient(indexerAddress, indexerToken)
	if err != nil {
		return
	}

	for numTx > 0 {
		// Query
		result, err := indexerClient.SearchForTransactions().CurrencyGreaterThan(minAmount).Limit(limit).NextToken(nextToken).Do(context.Background())
		if err != nil {
			return
		}

		transactions := result.Transactions
		numTx = len(transactions)
		nextToken = result.NextToken

		if numTx > 0 {
			// Print results
			JSON, err := json.MarshalIndent(transactions, "", "\t")
			if err != nil {
				return
			}
			fmt.Printf(string(JSON) + "\n")

			fmt.Println("End of page : ", numPages)
			fmt.Println("Transaction printed : ", len(transactions))
			fmt.Println("Next Token : ", nextToken)
			numPages++
		}
	}
}

File: examples/indexer/javascript/SearchTransactionsPaging.js
// SearchTransactionsPaging.js
// requires algosdk@1.6.1 or higher 
// verify installed version
// npm list algosdk

const algosdk = require('algosdk');
const indexer_token = "";
const indexer_server = "http://localhost";
const indexer_port = 8980;

// Instantiate the indexer client wrapper
let indexerClient = new algosdk.Indexer(indexer_token, indexer_server, indexer_port);

let nexttoken = "";
let numtx = 1;
// loop until there are no more transactions in the response
// for the limit(max limit is 1000  per request)    
(async () => {
    let min_amount = 100000000000000;
    let limit = 2;
    while (numtx > 0) {
        // execute code as long as condition is true
        let next_page = nexttoken;
        let response = await indexerClient.searchForTransactions()
            .limit(limit)
            .currencyGreaterThan(min_amount)
            .nextToken(next_page).do();
        let transactions = response['transactions'];
        numtx = transactions.length;
        if (numtx > 0)
        {
            nexttoken = response['next-token']; 
            console.log("Transaction Information: " + JSON.stringify(response, undefined, 2));           
        }
    }
})().catch(e => {
    console.log(e);
    console.trace();
});